src/index.ts

```diff
@@ -418,6 +415,13 @@ export const walletNameToAddressAndProfilePicture = async (
       walletName
     );
   }
+  if (!walletAddressAndProfilePicture.walletAddress && walletName.split(".").length >= 2) {
```

Comment:
Re `walletName.split(".").length >= 2)` - should this be a check earlier in the code, and we just return `null`? I don't know if any Solana wallet name type that is less than 2 parts.

Reply:
We have subdomains and so does Bonfida

though we don't expose creating a subdomain in the frontend, it is possible by calling the program directly

Reply:
Done in the call